### PR TITLE
Added separateFoldable

### PR DESCRIPTION
**Pull request by tkroman**
#2741 

Since this is a new method, mima is breaking but I haven't used it before, so not sure what to do here.

### CODE REVIEW

**Comment by kailuowang** [comment]
Thanks for the contribution. There is ways to get around breaking binary compatibility for this type of addition, but there are quite some hoops to jump through. The good news is in Q2 we are going to allow this type of changes on master with a new branching scheme. So if you can wait, it might be easier to just schedule this to cats 2.1. Otherwise I can guide you on how to avoid breaking BC on Scala 2.11. Let me know.

**Comment by tkroman** [comment]
@kailuowang I can definitely wait, but I'm OK with jumping through the hoops as well, so I defer to your judgement.

**Comment by kailuowang** [comment]
If you can wait, let's wait to avoid the extra boilerplate code on master.

**Comment by travisbrown** [comment]
@tkroman Do you mind merging master to fix the build? Otherwise looks good to me.

File: core/src/main/scala/cats/Alternative.scala [hunk 1]
```diff
@@ -41,6 +41,30 @@ import simulacrum.typeclass
...
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val l: List[Either[String, Int]] = List(Right(1), Left("error"))
+   * scala> Alternative[List].separate(l)
+   * res0: (List[String], List[Int]) = (List(error),List(1))
+   * }}}
+   */
+  def separateFoldable[G[_, _], A, B](fgab: F[G[A, B]])(implicit G: Bifoldable[G], FF: Foldable[F]): (F[A], F[B]) =
```
Comment (travisbrown) [inline core/src/main/scala/cats/Alternative.scala]: I'd prefer for the `Foldable` instance to come first in the list to match the order it appears in the argument, but I guess matching `separate`'s order for consistency is probably more important.

File: core/src/main/scala/cats/syntax/alternative.scala [hunk 2]
```diff
@@ -51,6 +56,25 @@ final class SeparateOps[F[_], G[_, _], A, B](private val fgab: F[G[A, B]]) exten
...
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val l: List[Either[String, Int]] = List(Right(1), Left("error"))
+   * scala> l.separateFoldable
+   * res0: (List[String], List[Int]) = (List(error),List(1))
+   * }}}
+   */
+  def separateFoldable(implicit
```
Comment (travisbrown) [inline core/src/main/scala/cats/syntax/alternative.scala]: This enrichment method could be added to `SeparateOps` since the target has the same type.

**Comment by travisbrown** [comment]
@tkroman Thanks! I think it'll still need merging with master to avoid the 2.11 bincompat check.

**Comment by travisbrown** [comment]
CI failure was Travis nonsense, just restarted that job.

**Comment by tkroman** [comment]
woohoo, thanks!

**Pull request merged**